src/TabView.tsx

```diff
@@ -87,9 +89,15 @@ const TabView = <Route extends BaseRoute>({
   // @ts-ignore
   const focusedKey = navigationState.routes[navigationState.index].key;
 
-  if (navigationState.routes.length > 6) {
-    throw new Error('TabView only supports up to 6 tabs');
-  }
+  const trimmedRoutes = useMemo(() => {
+    if (Platform.OS === 'android' && navigationState.routes.length > MAX_TABS) {
+      console.warn(`TabView only supports up to ${MAX_TABS} tabs on Android`);
+      return navigationState.routes.slice(0, MAX_TABS);
+    }
+    return navigationState.routes;
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
```

Comment:
You shouldn't ignore this rule. It should be recalculated when navigation state changes (for example someone dynamically adds a route)